tools/tslint-rules/noHostDecoratorInConcreteRule.ts

```diff
@@ -0,0 +1,36 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new Walker(sourceFile, this.getOptions()));
+    }
+}
+
+class Walker extends Lint.RuleWalker {
+    visitClassDeclaration(node: ts.ClassDeclaration) {
+      if (!node.modifiers || !this.getOptions().length) { return; }
+
+      // Do not check the class if its abstract.
+      if (!!node.modifiers.find(modifier => modifier.kind === ts.SyntaxKind.AbstractKeyword)) {
+        return;
+      }
+
+      node.members.forEach(el => {
+        if (el.decorators) {
+          el.decorators.forEach(decorator => {
+            const decoratorText: string = decorator.getChildAt(1).getText();
+            const matchedDecorator: string = this.getOptions().find(
+              (item: string) => decoratorText.startsWith(item));
+            if (!!matchedDecorator) {
+              this.addFailureFromStartToEnd(decorator.getChildAt(1).pos - 1, decorator.end,
```

Comment:
Not completely sure, but you should be able to use `this.addFailureAtNode(decorator, ...)` instead.

Reply:
I initially tried this, but the starting position for the decorator was the character after the previous node, which made the tslint annotation go crazy.